tools/shared/src/main/scala/org/scalajs/core/tools/jsdep/FlatJSDependency.scala

```diff
@@ -34,4 +34,9 @@ final class FlatJSDependency(
   @deprecated("Use relPath instead.", "0.6.1")
   val resourceName: String = relPath
 
+  override def toString(): String = s"""FlatJSDependency($origin, 
+      dependencies: ${dependencies.mkString(", ")}, 
+      commonJSName: $commonJSName, 
+      relPathMinified: $relPathMinified)""".stripMargin.replaceAll("\n", "")
```

Comment:
To comply with our [coding style](https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md), this should be formatted like this (also adding the missing `|`s):
```scala
  override def toString(): String = {
    s"""FlatJSDependency($origin, 
       |dependencies: ${dependencies.mkString(", ")}, 
       |commonJSName: $commonJSName, 
       |relPathMinified: $relPathMinified)""".stripMargin.replaceAll("\n", "")
  }
```
There is a problem with this formulation, though. It requires a space at the end of the lines, and that is not good at all, as most editor configs (including mine) strip trailing whitespace on saving the files. So it wouldn't be possible for me to maintain this formatting without switching editor.

The general structure that we already have in https://github.com/scala-js/scala-js/blob/47ef7be4d5e05287334b22374cab4beb10085600/tools/shared/src/main/scala/org/scalajs/core/tools/jsdep/JSDependency.scala#L83-L94 does not have this issue. I would suggest good old string concatenations instead:
```scala
  override def toString(): String = {
    s"FlatJSDependency($origin, " +
    s"dependencies: ${dependencies.mkString(", ")}, " +
    s"commonJSName: $commonJSName, " +
    s"relPathMinified: $relPathMinified)"
  }
```
which makes the spaces explicit before the `" +` ending each line.

Finally, why no `origin: ` before the `$origin` field, like the other fields?

(The same comments apply to the two other methods.)